### ISSUE TITLE
How do I set level using this SDK?

### ISSUE THREAD

**Issue by aguynamedben** [getsentry/sentry-electron#108]
How do I set `level` using this SDK? Here's what I'm trying:

```js
Sentry.captureException(error, { level: 'warning', extra: { account, firstAttempt } });
Sentry.captureMessage('foo bar 4', { level: 'warning' });
```

Both events make it into Sentry but with `level: error` instead of `level: warning` as expected. See event `ffe36b72ea924f578424adf88f65f8e6` in your system for details.

**Comment by HazAT** [comment]
Hey,
This works like explained here:
https://github.com/getsentry/sentry-electron/issues/90#issuecomment-405216122

Hope this helps.

**Comment by aguynamedben** [comment]
That syntax is ridiculous!!!

**Comment by aguynamedben** [comment]
My code:
```js
  Sentry.getDefaultHub().withScope(() => {
    Sentry.getDefaultHub().configureScope(scope => {
      scope.setExtra('level', 'info');
      Sentry.captureMessage(`Here is my message`);
    });
  });
```

In Sentry, in the properties at the top, it still says level is error:
<img width="459" alt="image 2018-09-05 at 1 32 57 am" src="https://user-images.githubusercontent.com/107841/45081287-bb1df900-b0ab-11e8-9a95-5fa12d3606d6.png">

In the Additional Data at the bottom, I see the value I passed:
<img width="320" alt="image 2018-09-05 at 1 34 13 am" src="https://user-images.githubusercontent.com/107841/45081355-e56fb680-b0ab-11e8-9550-00a8d569b3b3.png">

How do I pass `level` so that it's a first class citizen, not shoved down in Additional Data?

**Comment by HazAT** [comment]
@aguynamedben We are adding `level` for message again in `captureMessage` like:
```
Sentry.captureMessage('foo bar 4',  'warning' ); 
```
stay tuned for the next release.

**Comment by aguynamedben** [comment]
Cool, thanks for the update.

**Comment by aguynamedben** [comment]
Aside from level, I still believe this syntax is pretty laborious. I was just implementing logging of email addresses for failed login. What I tried first:

What I wish worked
```js
Sentry.captureException(error, { extra: { attemptedEmail: this.state.email } });
```
Side note: This old syntax use fails silently, which is confusing. If this syntax will not be supported, it would be helpful to log a warning message to remind developers of the new syntax.

What worked
```js
Sentry.getDefaultHub().withScope(() => {
  Sentry.getDefaultHub().configureScope((scope) => {
    scope.setExtra('attemptedEmail', this.state.email);
    Sentry.captureException(error);
  });
});
```

Error handling is already complex enough. Please support a simpler API along with the fancy one.

**Comment by HazAT** [comment]
@aguynamedben 
Many people already reported this, so we trimmed down the API by a bit
This is how it will work with the next release:
```javascript
Sentry.withScope(scope => {
    scope.setExtra('attemptedEmail', this.state.email);
    Sentry.captureException(error);
});
```
`withScope` now provides the scope, so a call to `configureScope` is optional.

About the warning, that's a good idea, will add it now.

**Comment by aguynamedben** [comment]
@HazAT That's a lot better, thanks.

**Comment by aguynamedben** [comment]
@HazAT sweet, looks like y'all released 0.10.0 of this library today, and it depends on 4.0.0 of sentry-javascript... I noticed in the [4.0.0 release notes](https://github.com/getsentry/sentry-javascript/releases) that the Sentry.withScope syntax is supported... good to try it out or should I hold off?

Thanks.

**Comment by HazAT** [comment]
@aguynamedben No it's good to use, go ahead.
